src/cat.c

```diff
@@ -323,7 +326,23 @@ static void ack_error(struct cat_object *self)
 {
         assert(self != NULL);
 
+        #if CONFIG_LIB_CAT_USRDATA
+        char *strbuf = (char *)malloc(512);
+        assert(NULL == strbuf);
+        memset(strbuf,0,ACKBUF_LEN);
+        if(get_cat_user_databuf_errorcode())
+        {
+                sprintf(strbuf,"ERROR%s",get_cat_user_databuf(CAT_USER_DATABUF_OPS_WRITE, CAT_USER_DATABUF_ACK_ERR));
```

Comment:
differentation (and customization) of the automatic ack/error response could be implemented more generic way, for e.g. with optional callbacks with return codes.